src/plugins/intel_gpu/src/graph/include/reshape_inst.h

```diff
@@ -28,6 +29,33 @@ struct typed_program_node<reshape> : public typed_program_node_base<reshape> {
 
     program_node& input() const { return get_dependency(0); }
 
+    bool is_runtime_propagatable_padding() const {
+        if (!input().is_type<crop>())
+            return false;
```

Comment:
Isn't this chainging the behavior? 
e.g., so far kv_cache with dynamic padding + unsqueeze reshape has been allowed. 
However with this change, won't it be prevented ? 
(i.e., previously is_in_place() was returning true if unsqueeze / squeeze mode. By using only this function, it will be prevented) 
What I suggested offline was to add return true if squeeze / unsqueeze here, (because propagate_pading is implemnented for those types), and then allow some special cases for base mode here...

Reply:
The intention of this function is to limit it to a specific case (crop + reshape) among cases for the base mode. The existing kv_cache + unsqueeze reshape case does not prevent buffer fusing in `is_in_place` as follow:
![image](https://github.com/openvinotoolkit/openvino/assets/47554900/0e843e54-d5f2-42ae-9b61-f08d526c7a7e)

Reply:
Generalize this function as your comment.